File: src/PL/ClientApp/src/app/_interfaces/basketModel.ts
import { TreeForBasketModel } from "./tree/treeForBasketModel";
import { ToyForBasketModel } from "./toy/toyForBasketModel";

export interface BasketModel {
    id: string;
    userName: string;
    trees: TreeForBasketModel[];
    toys: ToyForBasketModel[];
 }

File: src/PL/Models/BasketModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace PL.Models
{
    public class BasketModel
    {
        [Column("BasketId")]
        public Guid Id { get; set; }
        public string UserName { get; set; }

        public List<TreeForBasketModel> Trees { get; set; }
        public List<ToyForBasketModel> Toys { get; set; }
    }
}
